### PR TITLE
feat: update MVUX naming convention for generated classes

### LINKED ISSUE #2460: [MVUX] Naming changes for generated code
To avoid naming conflicts, we should looking into changing the generated code for Models to something other than the `BindableXYZ` naming pattern.

We should perhaps use a `ViewModel` suffix for the generated bindable version of the model

Will allow @dr1rrb / @francoistanguay to expand on their original thoughts on this

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): closes #2460 

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
Generated classes for MVUX would follow the pattern of XYZModel -> BIndableXYZModel

## What is the new behavior?
New generated classes will follow this new pattern XYZModel -> XYZViewModel

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md). (for bug fixes / features) 
- [x] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Extensions.Reactive.Generator/Bindables/ViewModelGenerator.cs
Comment (dr1rrb) [inline src/Uno.Extensions.Reactive.Generator/Bindables/ViewModelGenerator.cs:1]: I would also rename the file and the class of the old `BindableGenerator` to `ViewModelGenerator_1` and then this file to `ViewModelGenerator_2`.

BTW there is somehow a name clash between `ViewModelGenTool` and `VewModelGenerator` which are not generating the same thing ...

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.Extensions.Reactive.Generator/Bindables/ViewModelGenerator.cs
Comment (ajpinedam) [reply src/Uno.Extensions.Reactive.Generator/Bindables/ViewModelGenerator.cs:1]: Files renamed

**Comment by dansiegel** [comment]
It seems this PR is the culprit that is causing the builds to fail on the SDK update in the Templates repo. https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=137082&view=results

This is from a test PR, we've updated every other dependency except Extension and the build passes, as soon as we update to the build that was generated from this PR, the build fails.

**Comment by nickrandolph** [comment]
image

This isn't great. We should be still generating the BindableMainModel with it flagged as obsolete. 
@ajpinedam as fix, can you please reinstate the creation of the BidnableMainModel

**Pull request merged**